app/main.py

```diff
@@ -1,3 +1,64 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, int) or isinstance(other, float):
+            km = self.km + other
+        else:
+            km = self.km + other.km
+
+        return Distance(km)
+
+    def __iadd__(self, other: Distance | int | float) -> Distance:
+        if isinstance(other, int) or isinstance(other, float):
+            self.km += other
+        else:
+            self.km += other.km
+
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, int) or isinstance(other, float):
+            return self.km < other
+        else:
+            return self.km < other.km
+
+    def __gt__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, int) or isinstance(other, float):
+            return self.km > other
+        else:
+            return self.km > other.km
+
+    def __eq__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, int) or isinstance(other, float):
+            return self.km == other
+        else:
+            return self.km == other.km
+
+    def __le__(self, other: Distance | int | float) -> bool:
```

Comment:
You can use previously created magic method here(and in ge as well) to simplify your code. e.g. return not self > other